### PR TITLE
Feature/david/serp header experiment url replacement

### PR DESCRIPTION
**Pull request by malmstein**
Task/Issue URL: https://app.asana.com/0/488551667048375/1175100132980402

**Description**:
This experiment variant wants to show the full DDG url instead of the search query. In order to do that we'll show URL that is sent to the WebView.

**Steps to test this PR**:
1. Open app and enable variant `zg`
2. Make a search query in the native omnibar
3. After loading the page, the omnibar should show the full URL

1. Open app and enable variant `zg`
2. Make a search query in the SERP header
3. After loading the page, the omnibar should show the full URL

**Screenshots**:
Screenshot_1589201518

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)

### CODE REVIEW

File: app/src/main/java/com/duckduckgo/app/browser/BrowserTabViewModel.kt [hunk 1]
```diff
@@ -520,9 +530,17 @@ class BrowserTabViewModel(
 
     private fun omnibarTextForUrl(url: String?): String {
         if (url == null) return ""
-        if (duckDuckGoUrlDetector.isDuckDuckGoQueryUrl(url)) {
-            return duckDuckGoUrlDetector.extractQuery(url) ?: ""
+
+        return if (variantManager.getVariant().hasFeature(VariantManager.VariantFeature.SerpHeaderQueryReplacement)) {
```
Comment (subsymbolic) [inline app/src/main/java/com/duckduckgo/app/browser/BrowserTabViewModel.kt:534]: The nesting inside the return statement is hard to parse and is going to be a pain to unpick later if you need to remove this experiment. Can we update the new code to just be:

```
if (variantManager.getVariant().hasFeature(VariantManager.VariantFeature.SerpHeaderQueryReplacement)) {
   return url
}
```
And then we can continue with the original code as it was. If the experiment fails we then have a clear "block" of code we can delete.

**Comment by subsymbolic** [comment]
@malmstein let me know when https://github.com/duckduckgo/Android/pull/826 is merged and this branch is up-to-date with the changes from there and I will complete the review.

**Comment by malmstein** [comment]
@subsymbolic  this is good to go!
